dlib/cmake_utils/find_blas.cmake

```diff
@@ -1,480 +0,0 @@
-      set(blas_found 1)
-      set(lapack_found 1)
-      set(found_intel_mkl 1)
-      message(STATUS "Found Intel MKL BLAS/LAPACK library")
-
-      # Make sure the version of the Intel MKL we found is compatible with
-      # the compiler we are using.  One way to do this check is to see if we can
-      # link to it right now.
-      set(CMAKE_REQUIRED_LIBRARIES ${blas_libraries})
-      CHECK_FUNCTION_EXISTS(cblas_ddot MKL_HAS_CBLAS)
-      if (NOT MKL_HAS_CBLAS)
-         message("BLAS library does not have cblas symbols, so dlib will not use BLAS or LAPACK")
-         set(blas_found 0)
-         set(lapack_found 0)
-      endif()
-   endif()
-
-   if (found_intel_mkl AND mkl_include_dir)
-      set(found_intel_mkl_headers 1)
-   endif()
-
-endif()
-
-
-# When all else fails use CMake's built in functions to find BLAS and LAPACK
-if (NOT blas_found)
-   find_package(BLAS QUIET)
-   if (${BLAS_FOUND})
-      set(blas_libraries ${BLAS_LIBRARIES})      
-      set(blas_found 1)
-      if (NOT lapack_found)
-         find_package(LAPACK QUIET)
-         if (${LAPACK_FOUND})
-            set(lapack_libraries ${LAPACK_LIBRARIES})
-            set(lapack_found 1)
-         endif()
-      endif()
-   endif()
-endif()
-
-
-# If using lapack, determine whether to mangle functions
-if (lapack_found)
-   include(CheckFortranFunctionExists)
-   set(CMAKE_REQUIRED_LIBRARIES ${lapack_libraries})
-
-   check_function_exists("sgesv" LAPACK_FOUND_C_UNMANGLED)
-   check_function_exists("sgesv_" LAPACK_FOUND_C_MANGLED)
-   if (CMAKE_Fortran_COMPILER_LOADED)
-      check_fortran_function_exists("sgesv" LAPACK_FOUND_FORTRAN_UNMANGLED)
-      check_fortran_function_exists("sgesv_" LAPACK_FOUND_FORTRAN_MANGLED)
-   endif ()
-   if (LAPACK_FOUND_C_MANGLED OR LAPACK_FOUND_FORTRAN_MANGLED)
-      set(lapack_with_underscore 1)
-   elseif (LAPACK_FOUND_C_UNMANGLED OR LAPACK_FOUND_FORTRAN_UNMANGLED)
-      set(lapack_without_underscore 1)
-   endif ()
-endif()
-
-
-if (UNIX OR MINGW)
-   if (NOT blas_found)
-      message(" *****************************************************************************")
-      message(" *** No BLAS library found so using dlib's built in BLAS.  However, if you ***")
-      message(" *** install an optimized BLAS such as OpenBLAS or the Intel MKL your code ***")
-      message(" *** will run faster.  On Ubuntu you can install OpenBLAS by executing:    ***")
-      message(" ***    sudo apt-get install libopenblas-dev liblapack-dev                 ***")
-      message(" *** Or you can easily install OpenBLAS from source by downloading the     ***")
-      message(" *** source tar file from http://www.openblas.net, extracting it, and      ***")
-      message(" *** running:                                                              ***")
-      message(" ***    make; sudo make install                                            ***")
-      message(" *****************************************************************************")
```

Comment:
Totally keep these too.  Except tell people to install the IntelMKL now that that's installable via apt (I haven't tried installing it that way but I'm assuming it's good and about to try it :D )